src/graphql/schema/node/createNodeFieldEntry.ts

```diff
@@ -46,7 +46,7 @@ export default function createNodeFieldEntry (buildToken: BuildToken): [string,
       if (!primaryKey || !primaryKey.read)
         throw new Error(`Invalid id, no readable primary key on collection named '${collection.name}'.`)
 
-      const value = await primaryKey.read(context, keyValue)
+      const value = await primaryKey.read(context, keyValue, resolveInfo, getGqlOutputType(buildToken, collection.type).gqlType)
```

Comment:
Yeah, performance is still worse than I'd like... But a lot of these things are still voodoo to me 😳